File: DistinctArrayGenerator/DistinctArrayGenerator.java
import java.util.Arrays;

public class DistinctArrayGenerator {

    public static int[] solution(int N) {
        int[] array = new int[N];
        int total = 0;
        for(int i = 0; i < N-1; i++) {
            array[i] = i;
            total = total + i;
        }
        array[N-1] = -(total);
        return array;
    }

    public static void main(String[] args) {
        System.out.println(Arrays.toString(solution(4)));
        System.out.println("Sum: " + Arrays.stream(solution(4)).sum());
        System.out.println(Arrays.toString(solution(3)));
        System.out.println("Sum: " + Arrays.stream(solution(3)).sum());
    }
}

File: DistinctArrayGenerator/DistinctArrayGenerator.js
function solution(N) {
    let array = new Array(N);
    let total = 0;
    for(var i = 0; i < N-1; i++){
        array[i] = i;
        total = total + i;
    }
    array[N-1] = -(total);
    return array;
}

console.log(solution(4));
console.log("Sum: " + solution(4).reduce((accumulator, currentValue)=>accumulator+currentValue,0));
console.log(solution(3));
console.log("Sum: " + solution(3).reduce((accumulator, currentValue)=>accumulator+currentValue,0));